tests/fast-integration/skr-test/btp-manager-secret/index.js

```diff
@@ -0,0 +1,115 @@
+
+const reconciliationTimeout = 1000 * 70;
+let secretFromProvisioning;
+let modifiedSecret;
+
+function btpManagerSecretTest() {
+  describe('BTP Manager Secret Test', function() {
+    // Check if BTP Manager Secret with BTP Operator credentials is created properly
+    it('should check if Secret exists and contains the expected data keys', async function() {
+      secretFromProvisioning = await getSecret(secretName, ns);
+      checkSecretDataKeys(secretFromProvisioning);
+      modifiedSecret = JSON.parse(JSON.stringify(secretFromProvisioning));
+    });
+    // Check if the Secret contains expected values
+    it('should check if Secret data values match expected values', async function() {
+      const actualSecretData = await getSecretData(secretName, ns);
+      checkSecretDataValues(actualSecretData);
+    });
+    // Check if the Secret is properly reconciled after deletion
+    it('should check if Secret is reconciled after deletion', async function() {
+      console.log(`Deleting the "sap-btp-manager" Secret`);
+      await k8sDelete([secretFromProvisioning], ns);
+      console.log(`Waiting for the reconciliation for ${reconciliationTimeout} ms`);
+      await waitForSecret(secretName, ns, reconciliationTimeout);
+      console.log(`Secret has been re-created. Checking Secret's data`);
+      const actualSecret = await getSecret(secretName, ns);
+      checkSecretDataKeys(actualSecret);
+      const actualSecretData = await getSecretData(secretName, ns);
+      checkSecretDataValues(actualSecretData);
+      console.log(`Secret has been properly reconciled`);
+    });
+    // Check if the Secret is properly reconciled after being edited
+    it('should check if Secret is reconciled after being edited', async function() {
+      console.log(`Changing data in the "sap-btp-manager" Secret`);
+      prepareSecretForApply(modifiedSecret);
+      changeSecretData(modifiedSecret);
+      console.log(`Applying edited Secret`);
+      await k8sApply([modifiedSecret], ns);
+      console.log(`Waiting ${reconciliationTimeout} ms until edited Secret is created`);
+      await waitForSecret(secretName, ns, reconciliationTimeout);
+      let actualSecret = await getSecret(secretName, ns);
+      console.log(`Waiting for the reconciliation for ${reconciliationTimeout} ms`);
+      await waitForK8sObject(
+          `/api/v1/namespaces/${ns}/secrets`,
+          {},
+          (_type, _apiObj, watchObj) => {
+            return (
+              watchObj.object.metadata.name.includes(secretName) &&
+                        watchObj.object.metadata.resourceVersion !== actualSecret.metadata.resourceVersion
+            );
+          },
+          reconciliationTimeout,
+          `Waiting for ${secretName} Secret reconciliation timeout (${reconciliationTimeout} ms)`,
+      );
+      console.log(`Secret has been reconciled. Checking Secret's data`);
+      actualSecret = await getSecret(secretName, ns);
```

Comment:
This sequence of four lines is duplicated at least twice. So it could be extracted.
As a matter of fact, if we join the first two assertions we could use extracted function for the third time.

Reply:
I agree, we can wrap this sequence in a separate func to assert a Secret